livedata/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt

```diff
@@ -509,27 +509,22 @@ class ChannelControllerImpl(
         val mimeType = MimeTypeMap.getSingleton().getMimeTypeFromExtension(file.extension)
         val attachmentType = if (mimeType.isImageMimetype()) { "image" } else { "file" }
         val pathResult = if (attachmentType == "image") { sendImage(file) } else { sendFile(file) }
-        var newAttachment: Attachment
-        var uploadError: ChatError? = null
-
-        if (pathResult.isError) {
-            uploadError = pathResult.error()
-
-            newAttachment = attachment.copy(uploadState = Attachment.UploadState.Failed(uploadError))
-        } else {
-            val uploadPath = pathResult.data()
-            newAttachment = attachment.copy(
-                name = file.name,
-                fileSize = file.length().toInt(),
-                mimeType = mimeType?.toString() ?: "",
-                url = uploadPath,
-                uploadState = Attachment.UploadState.Success,
-                type = attachmentType
-            ).apply {
+        val url = if (pathResult.isError) null else pathResult.data()
+        val uploadState = if (pathResult.isError) Attachment.UploadState.Failed(pathResult.error()) else Attachment.UploadState.Success
+
+        var newAttachment = attachment.copy(
+            name = file.name,
+            fileSize = file.length().toInt(),
+            mimeType = mimeType ?: "",
+            url = url,
+            uploadState = uploadState,
+            type = attachmentType
+        ).apply {
+            url?.let {
                 if (attachmentType == "image") {
```

Comment:
hardcoded constant. What if make constant here and put to companion (for example)?